includes/wikia/models/DesignSystemGlobalFooterModel.class.php

```diff
@@ -462,78 +460,22 @@ public function getData() {
 		return $data;
 	}
 
-	private function getSitenameData() {
-		$wgSitenameForComscoreForWikiId = WikiFactory::getVarValueByName( 'wgSitenameForComscore', $this->wikiId );
+	abstract protected function getSitenameData();
 
-		if ( $wgSitenameForComscoreForWikiId ) {
-			$sitename = $wgSitenameForComscoreForWikiId;
-		} else {
-			$wgSitenameForWikiId = WikiFactory::getVarValueByName( 'wgSitename', $this->wikiId );
-
-			if ( $wgSitenameForWikiId ) {
-				$sitename = $wgSitenameForWikiId;
-			} else {
-				$sitename = $this->wg->Sitename;
-			}
-		}
-
-		return [
-			'type' => 'text',
-			'value' => $sitename
-		];
+	protected function getVerticalData() {
+		return null;
 	}
 
-	private function getVerticalData() {
-		$wikiFactoryInstance = WikiFactoryHub::getInstance();
-		$verticalData = $wikiFactoryInstance->getWikiVertical( $this->wikiId );
-
-		/**
-		 * We don't want to show vertical 'Other' instead we show vertical 'Lifestyle'
-		 * This is Comscore requirement
-		 */
-		if ( $verticalData['id'] == WikiFactoryHub::VERTICAL_ID_OTHER ) {
-			$verticalMessageKey = $wikiFactoryInstance->getAllVerticals()[WikiFactoryHub::VERTICAL_ID_LIFESTYLE]['short'];
-		} else {
-			$verticalMessageKey = $verticalData['short'];
-		}
-
-		/**
-		 * Possible outputs:
-		 * - global-footer-licensing-and-vertical-description-param-vertical-tv
-		 * - global-footer-licensing-and-vertical-description-param-vertical-games
-		 * - global-footer-licensing-and-vertical-description-param-vertical-lifestyle
-		 * - global-footer-licensing-and-vertical-description-param-vertical-books
-		 * - global-footer-licensing-and-vertical-description-param-vertical-music
-		 * - global-footer-licensing-and-vertical-description-param-vertical-comics
-		 * - global-footer-licensing-and-vertical-description-param-vertical-movies
-		 */
-		$verticalMessageKey = 'global-footer-licensing-and-vertical-description-param-vertical-' . $verticalMessageKey;
-
-		return [
-			'type' => 'translatable-text',
-			'key' => $verticalMessageKey
-		];
+	protected function getLicenseData() {
+		return null;
 	}
 
-	private function getLicenseData() {
-		$licenseText = WikiFactory::getVarValueByName( 'wgRightsText', $this->wikiId ) ?: $this->wg->RightsText;
-
-		return [
-			'type' => 'link-text',
-			'title' => [
-				'type' => 'text',
-				'value' => $licenseText
-			],
-			'href' => $this->getLicenseUrl()
-		];
-	}
-
-	private function getFandomOverview() {
+	protected function getFandomOverview() {
 		$out = [
 			'links' => [ ]
 		];
 
-		if ( $this->lang === static::DEFAULT_LANG ) {
+		if ( $this->lang === self::DEFAULT_LANG ) {
 			$out['links'] = [
 				[
 					'type' => 'link-branded',
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "link-branded" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)